buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy

```diff
@@ -839,7 +839,7 @@ class BeamModulePlugin implements Plugin<Project> {
         log4j2_slf4j_impl                           : "org.apache.logging.log4j:log4j-slf4j-impl:$log4j2_version",
         mockito_core                                : "org.mockito:mockito-core:3.7.7",
         mockito_inline                              : "org.mockito:mockito-inline:4.5.1",
-        mongo_java_driver                           : "org.mongodb:mongo-java-driver:3.12.11",
+        mongo_java_driver                           : "org.mongodb:mongodb-driver-legacy:5.3.1",
```

Comment:
To minimize the changes, I left the name of the library as is, but happy to change it to `mongodb-driver-legacy` if that makes sense.